frontend/docs/HOOKS.md

```
@@ -0,0 +1,168 @@
+### For New Components
+1. Create headless hook for business logic
+2. Create UI component that uses the hook
+3. Export both hook and component
+
+### For Existing Components
+1. Extract logic into headless hook
+2. Update component to use the hook
+3. Maintain backward compatibility
+4. Add new features via hook options
+
+## Future Hooks (Planned)
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)